### ISSUE TITLE
Link clickhouse failed.

### ISSUE THREAD

**Issue by ucasFL** [ClickHouse/ClickHouse#18827]
Make sure that `git diff` result is empty and you've just pulled fresh master. Try cleaning up cmake cache. Just in case, official build instructions are published here: https://clickhouse.yandex/docs/en/development/build/

**Operating system**
OS kind or distribution, specific version/release, non-standard kernel if any. If you are trying to build inside virtual machine, please mention it too.
CentOS 7
**Cmake version**
cmake version 3.19.2
**Ninja version**
1.10.2
**Compiler name and version**
clang 11.0
**Full cmake and/or ninja output**

ninja output
```
../base/daemon/BaseDaemon.cpp:102: error: undefined reference to 'sys_siglist'
../contrib/fmtlib-cmake/../fmtlib/include/fmt/core.h:1852: error: undefined reference to 'sys_siglist'
../base/daemon/BaseDaemon.cpp:98: error: undefined reference to 'sys_siglist'
../base/daemon/BaseDaemon.cpp:98: error: undefined reference to 'sys_siglist'
```

**Comment by azat** [comment]
@ucasFL Are you using bundled build?
There is a fix for unbundled build - #18764, but if this is a similar error you should have errors during compiling too not only during linking

**Comment by ucasFL** [comment]
> @ucasFL Are you using bundled build?

No. 

> There is a fix for unbundled build - #18764, but if this is a similar error you should have errors during compiling too not only during linking

I don't see other errors during build.

**Comment by azat** [comment]
>I don't see other errors during build.

I think that the problem you see only error during linking is because clickhouse is shipped with some version of glibc headers, that provides `sys_siglist`, but there is no such symbol in your libc, maybe it is too old or some patched version?

```sh
nm -D /path/to/libc.so.6 | fgrep sys_siglist
```

**Comment by ucasFL** [comment]
> > I don't see other errors during build.
> 
> I think that the problem you see only error during linking is because clickhouse is shipped with some version of glibc headers, that provides `sys_siglist`, but there is no such symbol in your libc, maybe it is too old or some patched version?
> 
> ```shell
> nm -D /path/to/libc.so.6 | fgrep sys_siglist
> ```

The symbol exists in my libc...

**Comment by ucasFL** [comment]
> > I don't see other errors during build.
> 
> I think that the problem you see only error during linking is because clickhouse is shipped with some version of glibc headers, that provides `sys_siglist`, but there is no such symbol in your libc, maybe it is too old or some patched version?
> 
> ```shell
> nm -D /path/to/libc.so.6 | fgrep sys_siglist
> ```

This is [cmake output](https://paste.ubuntu.com/p/Nt6DsHZNRR/)

**Comment by azat** [comment]
FWIW the issue should go away with #18858

**Comment by ucasFL** [comment]
> FWIW the issue should go away with #18858

Yes, I have checked it will be fixed with #18858
